plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSessionProperties.java

```diff
@@ -174,6 +168,11 @@ public DeltaLakeSessionProperties(
                         "Compression codec to use when writing new data files",
                         HiveCompressionCodec.class,
                         deltaLakeConfig.getCompressionCodec(),
+                        value -> {
+                            if (value == HiveCompressionCodec.LZ4) {
+                                throw new TrinoException(INVALID_SESSION_PROPERTY, "Unsupported codec: LZ4");
```

Comment:
No change requested. We could add an assertion to `DeltaLakeConfig` for denying LZ4 during initialization, but it might be overkill.

Reply:
Even without this check implemented, there is an exception about not supported compression during write. So I think we are safe here.